processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -81,12 +123,36 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 	hashableID := b.String()
 	identityBufferPool.Put(b)
 
+	var keep bool
 	s, ok := t.states.LoadOrStore(hashableID, &State{
 		PrevPoint: metricPoint,
 	})
 	if !ok {
-		return
+		switch t.initialValue {
+		case InitialValueDrop:
+			return
+		case InitialValueAuto:
+			if metricID.StartTimestamp < t.startTime || metricPoint.ObservedTimestamp == metricID.StartTimestamp {
+				return
+			}
+			out.StartTimestamp = metricID.StartTimestamp
+			keep = true
+		case InitialValueKeep:
+			keep = true
+		}
+		if keep {
```

Comment:
Looking at this again, it looks like we can get rid of the `keep` bool and do the `MetricType` switch after the `initialValue` switch always since `keep` is never false.

Reply:
nice observation, done